src/app/app.component.ts

```diff
@@ -136,6 +136,58 @@ export class AppComponent {
       { property: 'og:type', content: 'website' }
     ]);
 
+    this._flags.getFlagsFromService().pipe(
+      take(1),
+      map(flags => {
+        if (flags.enableOneTrust) {
+            window['OptanonWrapper'] = () => {
+                window['OneTrust'].InsertScript("//translate.google.com/translate_a/element.js?cb=googleTranslateElementInit", "body", null, null, "C0003");
+                window['OneTrust'].InsertScript("/assets/js/zendesk.js", "body", null, null, "C0003");
+                setTimeout( () => {
+                    window['zE'](() => {
+                        window['$zopim'](() => {
+                            window['$zopim'].livechat.setOnConnected(() => {
+                                if (this.shouldShowChat(window.location.hash)) {
+                                    window['zE']('webWidget', 'show');
+                                } else {
+                                    window['zE']('webWidget', 'hide');
+                                }
+                            });
+                        });
+                    });
+                }, 1000);
+
+                this._app.googleTranslateLoaded.next(true);
+            }
+        } else {
+            let googleTranslateScript = document.createElement('script');
+            googleTranslateScript.setAttribute('src','//translate.google.com/translate_a/element.js?cb=googleTranslateElementInit');
+            document.head.appendChild(googleTranslateScript);
+
+            let zendDeskScript = document.createElement('script');
+            zendDeskScript.setAttribute('src','/assets/js/zendesk.js');
+            document.head.appendChild(zendDeskScript);
+
+            this._app.googleTranslateLoaded.next(true);
+            
+            setTimeout(() => {
+                window['zE'](() => {
+                    window['$zopim'](() => {
+                        window['$zopim'].livechat.setOnConnected(() => {
+                            if (this.shouldShowChat(window.location.hash)) {
+                                window['zE']('webWidget', 'show');
+                            } else {
+                                window['zE']('webWidget', 'hide');
+                            }
+                        });
+                    });
+                });
+            }, 1000);
```

Comment:
These bits seem like they could be extracted to a single `initializeWidgets` or what have you, used in both the `if` and `else`. And the other parts could be extracted to a `loadWidgetScripts` or what have you. Then this function would look like:

```javascript
loadWidgetScripts();
initializeWidgets();
```

Reply:
@daneah changes made in the recent commit `001d1f6`.